independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -471,13 +474,24 @@ public Builder setAlternativePriorities(AlternativePriorities priorities) {
             return this;
         }
 
+        /**
+         * Specify the types that should be excluded from discovery.
+         * 
+         * @param predicate
+         * @return self
+         */
+        public Builder addExcludeType(Predicate<ClassInfo> predicate) {
+            this.excludeTypes.add(predicate);
+            return this;
+        }
+
         public BeanProcessor build() {
             return new BeanProcessor(name, index, additionalBeanDefiningAnnotations, output, sharedAnnotationLiterals,
                     reflectionRegistration, annotationTransformers, injectionPointTransformers, observerTransformers,
                     resourceAnnotations, beanRegistrars, observerRegistrars, contextRegistrars, beanDeploymentValidators,
                     applicationClassPredicate, removeUnusedBeans, removalExclusions, additionalStereotypes,
                     additionalInterceptorBindingRegistrars, transformUnproxyableClasses, jtaCapabilities, generateSources,
-                    allowMocking, alternativePriorities);
+                    allowMocking, alternativePriorities, excludeTypes);
```

Comment:
To avoid this crazy ctor you can make BeanProcessor(Builder builder) ctor instead in which init fields like `this.field = builder.field`.

Reply:
and/or apply some other processing.

Reply:
Yeah, the builder param is a good idea. I'll update the PR.